Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.mm

```diff
@@ -0,0 +1,324 @@
+        Vector<CoreIPCNSURLRequestData::Header> vector;
+        vector.reserveInitialCapacity(headerFields.count);
+        for (id key in headerFields) {
+            id value = [headerFields objectForKey:key];
+            if ([value isKindOfClass:[NSArray class]]) {
+                NSArray *valueArray = value;
+                Vector<String> valueVector;
+                valueVector.reserveInitialCapacity(valueArray.count);
+                for (id item in valueArray) {
+                    if ([item isKindOfClass:[NSString class]]) {
+                        NSString *s = item;
+                        valueVector.append(s);
+                    }
+                }
+                vector.append({ key, valueVector });
+            }
+            if ([value isKindOfClass:[NSString class]])
+                vector.append({ key , value });
+        }
+        vector.shrinkToFit();
+        m_data.headerFields = WTFMove(vector);
+    }
+
+    SET_NSURLREQUESTDATA(body, NSData, CoreIPCData);
+
+    NSArray *bodyParts = dict[@"bodyParts"];
+    if (bodyParts != nil && [bodyParts isKindOfClass:[NSArray class]]) {
+        Vector<CoreIPCNSURLRequestData::BodyParts> vector;
+        vector.reserveInitialCapacity(bodyParts.count);
+        for (id element in bodyParts) {
+            if ([element isKindOfClass:[NSString class]]) {
+                CoreIPCString tooAdd(element);
+                vector.append(WTFMove(tooAdd));
+            }
+            if ([element isKindOfClass:[NSData class]]) {
+                CoreIPCData tooAdd(element);
+                vector.append(WTFMove(tooAdd));
+            }
+        }
+        vector.shrinkToFit();
+        m_data.bodyParts = WTFMove(vector);
+    }
+
+    SET_NSURLREQUESTDATA_PRIMITIVE(startTimeoutTime, NSNumber, double);
+    SET_NSURLREQUESTDATA_PRIMITIVE(requiresShortConnectionTimeout, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(payloadTransmissionTimeout, NSNumber, double);
+
+    id allowedProtocolTypes = dict[@"allowedProtocolTypes"];
+    if (allowedProtocolTypes != nil && [allowedProtocolTypes isKindOfClass:[NSNumber class]]) {
+        auto val = [allowedProtocolTypes unsignedCharValue];
+        m_data.allowedProtocolTypes = static_cast<NSURLRequestAllowedProtocolTypes>(val);
+    }
+
+    SET_NSURLREQUESTDATA(boundInterfaceIdentifier, NSString, CoreIPCString);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsExpensiveNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsConstrainedNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsUCA, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(assumesHTTP3Capable, NSNumber, bool);
+
+    id attribution = dict[@"attribution"];
+    if (attribution != nil && [attribution isKindOfClass:[NSNumber class]]) {
+        auto val = [allowedProtocolTypes unsignedCharValue];
+        m_data.attribution = static_cast<NSURLRequestAttribution>(val);
```

Comment:
For these cases where we cast to an enum, let's make sure it's a valid value first using isValidEnum.